bfs.py

```diff
@@ -19,12 +19,37 @@ def bfs(self, source):
                 if not neighbor.visited:
                     neighbor.cost = candidate.cost + 1
                     neighbor.visited = True
+                    neighbor.parent = candidate
                     candidates.append(neighbor)
+                    print(neighbor.name, "neighbor", candidate.name)
                     print("Added", neighbor.name, "to candidates. Cost:", neighbor.cost)
                 else:
                     print(neighbor.name, "already visited! Skipping...")
             print("----------------------------------------")
         print("---------------End of BFS---------------")
+        
+    def path_finder(self, source, destination):
+        x = 1
+        print("******************** By the use of bfs, we are initiating the path search from", source.name, "to", destination.name, "and the travel cost*******************")
+        self.bfs(source)
+        traversal = []
+        traversal.append(destination.name)
+        while destination.parent != None:      
+            traversal.append(destination.parent.name)
+            destination = destination.parent
+        traversal.reverse()
+        print(" ")
```

Comment:
Why do we need this print?

Reply:
It was used to make the output look a bit more readable.